src/ucp/wireup/wireup_ep.c

```diff
@@ -441,21 +515,51 @@ ssize_t ucp_wireup_ep_sockaddr_fill_private_data(void *arg, const char *dev_name
 
     conn_priv_len = sizeof(*conn_priv) + address_length;
 
-    /* check private data length limitation */
-    wiface = &worker->ifaces[sockaddr_rsc];
-    if (conn_priv_len > wiface->attr.max_conn_priv) {
-        ucs_error("worker address information (%zu) exceeds max_priv on "
-                   UCT_TL_RESOURCE_DESC_FMT" (%zu)", conn_priv_len,
-                   UCT_TL_RESOURCE_DESC_ARG(&context->tl_rscs[sockaddr_rsc].tl_rsc),
-                   wiface->attr.max_conn_priv);
-        status = UCS_ERR_BUFFER_TOO_SMALL;
-        goto err_free_address;
-    }
-
     /* pack private data */
     conn_priv->err_mode = ucp_ep_config(ucp_ep)->key.err_mode;
     conn_priv->ep_ptr   = (uintptr_t)ucp_ep;
-    memcpy(conn_priv + 1, worker_address, address_length);
+
+    wiface = &worker->ifaces[sockaddr_rsc];
+
+    /* check private data length limitation */
+    if (conn_priv_len > wiface->attr.max_conn_priv) {
+
+        /* since the full worker address is too large to fit into the trasnport's
+         * private data, try to pack resources to pass partial address */
+        status = ucp_wireup_ep_pack_always_supported_rscs(worker, dev_name, &tl_id,
```

Comment:
pls add ucs_trace() message that we use partial addr with all details (length, which tl is used, ...)